### ISSUE TITLE
Question about calculating subsampled mask after TDNN

### ISSUE THREAD

**Issue by yjiangling** [espnet/espnet#3659]
Hi,

     I read the code of tdnn.py in "espnet/nets/pytorch_backend/transducer/tdnn.py", and the content about re_calculate mask after TDNN layer like this:

`    def create_outputs(
        self,
        sequence: torch.Tensor,
        pos_emb: torch.Tensor,
        mask: torch.Tensor,
        bidir_pos_emb: bool,
    ) -> Tuple[Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]], torch.Tensor]:
        """Create outputs with subsampled version of pos_emb and masks.

        Args:
            sequence: TDNN output sequences. (B, sub(T), D_out)
            pos_emb: Input positional embedding.
                     (B, T, att_dim) or (B, 2 * (T - 1), D_att)
            mask: Mask of TDNN input sequences. (B, 1, T)
            bidir_pos_emb: Whether to use bidirectional positional embedding.

        Returns:
            sequence: TDNN output sequences. (B, sub(T), D_out)
            pos_emb: Output positional embedding.
                     (B, sub(T), D_att) or (B, 2 * (sub(T) - 1), D_att)
            mask: Mask of TDNN output sequences. (B, 1, sub(T))

        """
        **sub = (self.ctx_size - 1) * self.dilation

        if mask is not None:
            if sub != 0:
                mask = mask[:, :, :-sub]

            mask = mask[:, :, :: self.stride]**

        if pos_emb is not None:
            # If the bidirect_pos is true, the pos_emb will include both positive and
            # negative embeddings. Refer to https://github.com/espnet/espnet/pull/2816.
            if bidir_pos_emb:
                pos_emb_positive = pos_emb[:, : pos_emb.size(1) // 2 + 1, :]
                pos_emb_negative = pos_emb[:, pos_emb.size(1) // 2 :, :]

                if sub != 0:
                    pos_emb_positive = pos_emb_positive[:, :-sub, :]
                    pos_emb_negative = pos_emb_negative[:, :-sub, :]

                pos_emb_positive = pos_emb_positive[:, :: self.stride, :]
                pos_emb_negative = pos_emb_negative[:, :: self.stride, :]

                pos_emb = torch.cat(
                    [pos_emb_positive, pos_emb_negative[:, 1:, :]], dim=1
                )
            else:
                if sub != 0:
                    pos_emb = pos_emb[:, :-sub, :]

                pos_emb = pos_emb[:, :: self.stride, :]

            return (sequence, pos_emb), mask

        return sequence, mask`

For example, if the original mask is         
masks = [[0, 0, 0, 0 ,0, 0],
                 [0, 0, 0, 0, 1, 1],
                 [0, 0, 0, 1, 1, 1]]

and the TDNN: ctx_size = 3, dilation = 1, stride = 1, the subsampled mask will be like this
subsampled_masks = [[0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 1]] instead of [[0, 0, 0, 0], [0, 0, 1, 1], [0, 1, 1, 1]].
This mean that the shorter sample in the batch will use the padding content to conduct self attention?
Is my understanding correct？ Looking forward to your reply. Thanks a lot!
